core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/sync/operationresult/CmSyncOperationResultEvaluatorService.java

```diff
@@ -0,0 +1,57 @@
+package com.sequenceiq.cloudbreak.service.upgrade.sync.operationresult;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+
+import com.google.common.collect.Sets;
+import com.sequenceiq.cloudbreak.service.upgrade.sync.ParcelInfo;
+
+@Service
+public class CmSyncOperationResultEvaluatorService {
+
+    public void evaluateParcelSync(Optional<CmParcelSyncOperationResult> cmParcelSyncOperationResultOptional,
+            CmSyncOperationSummary.Builder cmSyncOperationSummaryBuilder) {
+        if (cmParcelSyncOperationResultOptional.isEmpty() || cmParcelSyncOperationResultOptional.get().getInstalledParcels().isEmpty()) {
+            cmSyncOperationSummaryBuilder.withError("CM parcel sync failed, it was not possible to retrieve installed parcel versions from the CM server.");
+        } else {
+            CmParcelSyncOperationResult cmParcelSyncOperationResult = cmParcelSyncOperationResultOptional.get();
+            Set<String> installedParcelNames = cmParcelSyncOperationResult.getInstalledParcels().stream().map(ParcelInfo::getName).collect(Collectors.toSet());
+            Set<String> foundCmProductNames = cmParcelSyncOperationResult.getFoundCmProducts().stream().map(cmp -> cmp.getName()).collect(Collectors.toSet());
+            Set<String> notFoundProductNames = Sets.difference(installedParcelNames, foundCmProductNames);
+            if (notFoundProductNames.isEmpty()) {
+                String message = String.format("Successfully synced following parcel versions from CM server: %s", installedParcelNames);
+                cmSyncOperationSummaryBuilder.withSuccess(message);
+            } else {
+                String message;
+                if (notFoundProductNames.size() == installedParcelNames.size()) {
+                    message = String.format("The version of parcels could not be synced from CM server: %s.", notFoundProductNames);
+                } else {
+                    message = String.format("The version of some parcels could not be synced from CM server: %s. Parcel versions successfully synced: %s ",
+                            notFoundProductNames, foundCmProductNames);
+                }
```

Comment:
I have 2 suggestion for this:
- move out the if-else to a method so you don't have to declare the message empty beforehand
- or just keep the second message, which would simplify this a bit